File: qyMessengerApp/qyMessengerApp.Android/public/confReq.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using qyMessengerApp.Droid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace qy.Droid
{
	partial class qyFuncs_Droid
{
		public static unsafe int sendConfReq(void* p, object hDlgTalk, QY_MESSENGER_ID idInfo_conf, ushort usOp, ushort usAvLevel, int conf_iFourcc)
		{
			int iErr = -1;
			CCtxQyMc pQyMc = g.g_pQyMc;
			CCtxQmc pProcInfo = pQyMc.get_pProcInfo();//(  MC_VAR_isCli  *  )m_var.pMisCnt->pProcInfoParam;
			MIS_CNT pMisCnt = pProcInfo.pMisCnt;
			MainActivity mainActivity = (MainActivity)pQyMc.gui.hMainWnd;

			//
			using (CHelp_getDlgTalkVar help_getDlgTalkVar = new CHelp_getDlgTalkVar()) {
				DLG_TALK_var pm_var = (DLG_TALK_var)help_getDlgTalkVar.getVar(hDlgTalk);
				if (null==pm_var) return -1;
				DLG_TALK_var  m_var = pm_var;

				int lenInBytes = 0;

				//
				QY_MESSENGER_ID idInfo_to;

				//
				if (0==idInfo_conf.ui64Id) return -1;

				//
				CONF_req req;
				req.uiType = Consts.CONST_imCommType_confReq;
				req.idInfo_imGrp_related = idInfo_conf;
				req.usOp = usOp;
				req.policy.usAvLevel = usAvLevel;
				req.policy.conf_iFourcc = conf_iFourcc;

				//
				lenInBytes = sizeof(CONF_req);
				//					
				Int64 tStartTran;
				uint uiTranNo;
				qyFuncs.MACRO_prepareForTran(out tStartTran,out uiTranNo);


				//	
				pProcInfo.postMsg2Mgr_mc(mainActivity.var_common.ctxCaller,pMisCnt, null, Consts.CONST_misMsgType_req, 0, Consts.CONST_qyCmd_sendReq, tStartTran, uiTranNo, 0, (byte*)&req, (uint)lenInBytes, null, null, 0, null, false);
			}

			iErr = 0;
		errLabel:

			//
			string str;
			str=string.Format("sendReq: conf {0} {1} ", idInfo_conf.ui64Id, usOp);
			qy.TMP_buf5 buf5;
			qyFuncs.iFourcc2Str(conf_iFourcc, buf5.buf, Consts.cntof_buf5);
			str += qyFuncs.mytoString(buf5.buf);
			//_sntprintf(tBuf, mycountof(tBuf), _T("%s conf_iFourcc %S"), tBuf, buf);
			qyFuncs.showInfo_open(0, null, str);

			//
			return iErr;
		}


	}
}

File: srcProjs/srcRoot/src6/qyMcMain_isCli_common/qyMcMainPublic_isCli/public_isCli/confReq.cpp

#include	"stdafx.h"
#include	<time.h>
#include	"qyMcMainCommon.h"

#include	"qmcCommFunc_mis.h"
#include	"help_getDlgTalkVar.h"
#include	"dlgTalkProc.h"
#include	"qyMsgFunc.h"

//
 //
  int  set_doConf(  CONF_req  *  pConfReq,  LPCTSTR  hint  )
  {
	  CCtxQmc  *  pProcInfo  =  QY_GET_procInfo_isCli(  );
	  if  (  !pConfReq  )  return  -1;

	  if  (  !hint  )  hint  =  _T(  ""  );

	  //
	  if  (  pProcInfo->av.doConf.confReq.idInfo_imGrp_related.ui64Id  )  {
		  showInfo_open0(  0,  0,  _T(  "Serious error: set_doConf, idInfo_conf is not 0"  )  );
	  }

	  //
	  pProcInfo->av.doConf.confReq  =  *pConfReq;

	  //


	  //
	  TCHAR  tBuf[128];
	  char  buf[128];
	  iFourcc2Str(  pConfReq->policy.talkerPolicy.conf_iFourcc,  buf,  mycountof(  buf  )  );
	  //
	  _sntprintf(  tBuf,  mycountof(  tBuf  ),  _T(  "set_doConf ok, confReq: idInfo_conf %I64u,  level %d, conf_iFourcc %S, 2Streams %d, ucHardwareAccl %d, %s"  ),  
		  pConfReq->idInfo_imGrp_related.ui64Id,  
		  (int)pConfReq->policy.usAvLevel,
		  buf,
		  (int)pConfReq->hgInfo.bShare_2ndStream,
		  (int)pConfReq->ucHardwareAccl,
		  hint  );
	  showInfo_open0(  0,  0,  tBuf  );
	  _sntprintf(tBuf, mycountof(tBuf), _T("confReq: meetingId_hg %I64u"), pConfReq->hgInfo.ui64MeetingId_hg);
	  showInfo_open0(0, 0, tBuf);
	  _sntprintf(tBuf, mycountof(tBuf), _T("confReq: initiator %I64u"), pConfReq->idInfo_initiator.ui64Id);
	  showInfo_open0(0, 0, tBuf);

	  //
	  return  0;
  }

  int  clear_doConf(  QY_MESSENGER_ID  idInfo_conf,  LPCTSTR  hint  )
  {
	  CCtxQmc  *  pProcInfo  =  QY_GET_procInfo_isCli(  );
	  TCHAR  tBuf[128];

	  if  (  !hint  )  hint  =  _T(  ""  );

	  //
		  _sntprintf(  tBuf,  mycountof(  tBuf  ),  _T(  "Serious error: clear_doConf: idInfo_conf %I64u != doConf.%I64u"  ),  idInfo_conf.ui64Id,  pProcInfo->av.doConf.confReq.idInfo_imGrp_related.ui64Id  );
	  if  (  idInfo_conf.ui64Id  !=  pProcInfo->av.doConf.confReq.idInfo_imGrp_related.ui64Id  )  {
		  showInfo_open0(  0,  0,  tBuf  );
	  }


	  //
	  memset(  &pProcInfo->av.doConf,  0,  sizeof(  pProcInfo->av.doConf  )  );

	  _sntprintf(  tBuf,  mycountof(  tBuf  ),  _T(  "set_doConf ok, %s"  ),  hint  );
	  showInfo_open0(  0,  0,  tBuf  );

	  return  0;
  }




//
//int  sendConfReq(  void  *  p,  HWND  hDlgTalk,  QY_MESSENGER_ID  idInfo_conf,  unsigned  short  usOp,  unsigned  short  usAvLevel,  int  conf_iFourcc,  unsigned  short  conf_usMaxSpeakers  )
 int  sendConfReq(void* p, HWND  hDlgTalk, QY_MESSENGER_ID  idInfo_conf, unsigned  short  usOp, unsigned  short  usAvLevel, unsigned  short  conf_usMaxSpeakers, TALKER_policy* pTalkerPolicy, CONF_hg_info * pHgInfo)
{
	int  iErr  =  -1;
		CHelp_getDlgTalkVar				help_getDlgTalkVar;
	DLG_TALK_var				*	pm_var		=		(  DLG_TALK_var	*	)help_getDlgTalkVar.getVar(  hDlgTalk  );
	if  (  !pm_var  )  return  -1;
	DLG_TALK_var				&	m_var		=		*pm_var;
	MC_VAR_isCli					*	pProcInfo	=		QY_GET_procInfo_isCli(  );//(  MC_VAR_isCli  *  )m_var.pMisCnt->pProcInfoParam;

	int  lenInBytes  =  0;

	//
	QY_MESSENGER_ID		idInfo_to;

	//
	TALKER_policy talkerPolicy;
	memset(&talkerPolicy, 0, sizeof(talkerPolicy));
	if (pTalkerPolicy) talkerPolicy = *pTalkerPolicy;

	//
	if  (  !idInfo_conf.ui64Id  )  return  -1;
	
	//
	CONF_req  req;
	memset(&req, 0, sizeof(req));
	req.uiType  =  CONST_imCommType_confReq;
	req.idInfo_imGrp_related=idInfo_conf;
	req.usOp  =  usOp;
	req.policy.usAvLevel  =  usAvLevel;
	//
	req.policy.usMaxSpeakers = conf_usMaxSpeakers;
	//
	//req.policy.talkerPolicy.conf_iFourcc  =  pTalkerPolicy->conf_iFourcc;
	req.policy.talkerPolicy = talkerPolicy;

	//
	if (pHgInfo) {
		req.hgInfo = *pHgInfo;
	}

	//
	lenInBytes  =  sizeof(  req  );			
	//					
	MACRO_prepareForTran(  );

	
	//	
	pProcInfo->postMsg2Mgr_mc(  m_var.pMisCnt,  NULL,  CONST_misMsgType_req,  0,  CONST_qyCmd_sendReq,  tStartTran,  uiTranNo,  0,  (  char  *  )&req,  lenInBytes,  NULL,  0,  0,  NULL,  FALSE  );
	

	iErr  =  0;
errLabel:

	//
	TCHAR  tBuf[128];
	_sntprintf(  tBuf,  mycountof(  tBuf  ),  _T(  "sendReq: conf %I64u %s"  ),  idInfo_conf.ui64Id,  qyGetDesByType1(  CONST_imOpTable_en,  usOp  )  );
	char  buf[128];
	iFourcc2Str(  talkerPolicy.conf_iFourcc,  buf,  mycountof(  buf  )  );
	_sntprintf(  tBuf,  mycountof(  tBuf  ),  _T(  "%s conf_iFourcc %S"  ),  tBuf,  buf  );
	showInfo_open0(  0,  0,  tBuf  );

	//
	return  iErr;
}

